src/jdns/jdns.c

```diff
@@ -2067,6 +2067,10 @@ int _unicast_do_writes(jdns_session_t *s, int now)
 		}
 
 		// in theory, it is not possible for 'ns' to be null here
+		if (ns == 0)
+		{
+			continue; // but lets make sure it will not stab us in the back
```

Comment:
did it ever crash for you here? From what I can see  in the code it is really not possible for 'ns' to be null.